classes/07class/exercises/c07-serverless01/alanlima/terraform/main.tf

```diff
@@ -0,0 +1,81 @@
+data "archive_file" "customers_lambda" {
+  type        = "zip"
+  output_path = "${path.module}/files/customers_lambda.zip"
+
+  source {
+    content  = file("../src/common.py")
+    filename = "common.py"
+  }
+
+  source {
+    content  = file("../src/main.py")
+    filename = "main.py"
+  }
+}
+
+resource "aws_ssm_parameter" "db_name" {
+  name  = "/${var.project}/DB_NAME"
+  type  = "String"
+  value = var.db_name
+  tags  = var.common_tags
+}
+
+resource "aws_ssm_parameter" "api_key" {
+  name  = "/${var.project}/API_KEY"
+  type  = "SecureString"
+  value = aws_api_gateway_api_key.default.value
+  tags  = var.common_tags
+}
+
+resource "aws_kms_key" "this" {
+  description             = "${var.project} - DB Encrypt Key"
+  deletion_window_in_days = 7
+  tags                    = var.common_tags
+}
+
+# data "aws_kms_key" "this" {
+#   key_id = "0353c4e7-5cb9-473f-a3ae-f01795dab9a5"
+# }
+
+resource "aws_dynamodb_table" "this" {
+  name             = var.db_name
+  billing_mode     = "PROVISIONED"
+  write_capacity   = 5
+  read_capacity    = 5
```

Comment:
As we don't know the volume, better to pay-as-you-go. Use `billing_mode= PAY_PER_REQUEST` and don't specify WCU and RCU